controllers/cronset_controller.go

```diff
@@ -115,3 +129,61 @@ func (r *CronSetReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 	return nil
 }
+
+func (r *CronSetReconciler) applyCronJob(ctx context.Context, cronSet *batchv1alpha1.CronSet, nodeName string) error {
+	cronJobName := generateCronJobName(cronSet.Name, nodeName)
+	cronJobKey := metav1.ObjectMeta{
+		Name:      cronJobName,
+		Namespace: cronSet.Namespace,
+	}
+	cronJob := &batchv1.CronJob{
+		ObjectMeta: cronJobKey,
+	}
+
+	_, err := ctrl.CreateOrUpdate(ctx, r.Client, cronJob, func() error {
+		updateCronJobSpec(cronJob, cronSet, nodeName)
+		return controllerutil.SetControllerReference(cronSet, cronJob, r.Scheme)
+	})
+	if err != nil && errors.IsInvalid(err) {
+		_ = r.Delete(ctx, &batchv1.CronJob{ObjectMeta: cronJobKey}, client.PropagationPolicy("Background"))
+		return err
+	}
```

Comment:
it would better to add logs when occuring erros to debug well. I think this code totally doesn't have enough log for users.

we could do it next step after you will merge all you have already developed.